File: Javascript/Arrays/Problem7.js

// Move zero to last




let arr=[1 ,0 ,2 ,3 ,0 ,4 ,0 ,1];

let j=-1;

for(let i=0;i<arr.length;i++){
  if(arr[i]==0){
    j=i;
    break;
  }
}


for(let i=j+1;i<arr.length;i++){
  if(arr[i]!==0){
    let temp=arr[i];
    arr[i]=arr[j];
    arr[j]=temp;
    j++;
  }
}

console.log(arr)

File: C++/Arrays/Problem7.cpp
// Move zero last


#include <iostream>
using namespace std;


void MoveZero(int arr[],int n){
  
  int j=-1;
  
  for(int i=0;i<n;i++){
    if(arr[i]==0){
      j=i;
      break;
    }
  }
  
  for(int i=j+1;i<n;i++){
    if(arr[i]!=0){
      int temp=arr[i];
      arr[i]=arr[j];
      arr[j]=temp;
      j++;
    }
  }
  
}

int main() 
{
    int n=6;
    int arr[n]={1,2,0,3,0,4};
    
    MoveZero(arr,n);
    
   
    

    
    for(int i=0;i<n;i++){
      std::cout << arr[i]<<" ";
    }
}